modules/architecture-overview.adoc

```diff
@@ -0,0 +1,70 @@
+// Module included in the following assemblies:
+//
+// * architecture/architecture.adoc
+
+[id='architecture-overview-{context}']
+= Architecture overview
+
+Like {product-title} v3, {product-title} v4 is a layered system designed to
+expose underlying container images and Kubernetes concepts as accurately as
+possible, with a focus on easy composition of applications by a developer.
+For example, after your cluster is running, install Ruby, push code, and add
+MySQL to the app.
+
+A major difference between {product-title} v3 and v4 is that v4 uses Operators
+as both the fundamental unit of the product and an option for easily deploying
+and managing utilities that your apps use.
+
+{product-title} has an Operator-based architecture of smaller, decoupled units
+that work together. It runs on top of a Kubernetes cluster, with data about the
+objects stored in etcd, a reliable clustered key-value store. Those services are
+broken down by function:
+
+* Operators, which run the core {product-title} services.
+* REST APIs, which expose each of the core objects:
+** Containers and images, which are the building blocks for deploying your
+applications.
+** Pods and services, which containers use to communicate with each other and
+proxy connections.
+** Projects and users, which provide the space and means for communities to
+organize and manage their content together.
+** Builds and image streams] allow you to
+build working images and react to new images.
+** Deployments, which expand support for the software development and deployment
+lifecycle.
+** Routes, which announce your service to the world.
+** Templates, which allow you to simultaneously create many objects that are
+based on customized parameters.
+* Controllers, which read those REST APIs, apply changes to other objects, and
+report status or write back to the object.
+
```

Comment:
This needs to be split out into more of separate section, probably leading with “OpenShift and Kuberneres are API first systems that take application author intent and ensure that intent is maintained, even if hardware or software fails” and then moving all these details into sub bullets.

Reply:
(Everything to bottom)

Then we need to go into more detail about the parts we ship and how they work together.  Probably talk about config as API (all management config is available as an API), how we do updates, and how the monitoring stack works with across the board to give admins visibility into the state of their applications.